src/main/java/graphql/schema/visibility/NoIntrospectionGraphqlFieldVisibility.java

```diff
@@ -10,12 +10,17 @@
 
 /**
  * This field visibility will prevent Introspection queries from being performed.  Technically this puts your
- * system in contravention of the specification - http://facebook.github.io/graphql/#sec-Introspection but some
- * production systems want this lock down in place.
+ * system in contravention of <a href="https://spec.graphql.org/October2021/#sec-Introspection">the specification</a>
+ * but some production systems want this lock down in place.
+ *
+ * @deprecated This is no longer the best way to prevent Introspection - {@link graphql.introspection.Introspection#enabledJvmWide(boolean)}
+ * can be used instead
  */
 @PublicApi
+@Deprecated // (since = "2024-03-16")
 public class NoIntrospectionGraphqlFieldVisibility implements GraphqlFieldVisibility {
 
+    @Deprecated // (since = "2024-03-16")
```

Comment:
Java 8 special: since is not an argument for the Deprecated annotation, so it's a comment

This was added in Java 9. v21 uses Java 11.

Reply:
nice